editor/editor_dock_manager.cpp

```diff
@@ -818,14 +826,19 @@ void EditorDockManager::register_dock_slot(DockSlot p_dock_slot, TabContainer *p
 	p_tab_container->set_custom_minimum_size(Size2(170, 0) * EDSCALE);
 	p_tab_container->set_v_size_flags(Control::SIZE_EXPAND_FILL);
 	p_tab_container->set_popup(dock_context_popup);
-	p_tab_container->connect("pre_popup_pressed", callable_mp(dock_context_popup, &DockContextPopup::select_current_dock_in_dock_slot).bind(p_dock_slot));
 	p_tab_container->set_drag_to_rearrange_enabled(true);
 	p_tab_container->set_tabs_rearrange_group(1);
+	p_tab_container->set_use_hidden_tabs_for_min_size(true);
+
+	p_tab_container->connect("pre_popup_pressed", callable_mp(dock_context_popup, &DockContextPopup::select_current_dock_in_dock_slot).bind(p_dock_slot));
 	p_tab_container->connect("tab_changed", callable_mp(this, &EditorDockManager::_update_layout).unbind(1));
 	p_tab_container->connect("active_tab_rearranged", callable_mp(this, &EditorDockManager::_update_layout).unbind(1));
 	p_tab_container->connect("child_order_changed", callable_mp(this, &EditorDockManager::_dock_container_update_visibility).bind(p_tab_container));
-	p_tab_container->set_use_hidden_tabs_for_min_size(true);
-	p_tab_container->get_tab_bar()->connect(SceneStringName(gui_input), callable_mp(this, &EditorDockManager::_dock_container_gui_input).bind(p_tab_container));
+
+	TabBar *tab_bar = p_tab_container->get_tab_bar();
+	tab_bar->connect(SceneStringName(gui_input), callable_mp(this, &EditorDockManager::_dock_container_gui_input).bind(p_tab_container));
+	tab_bar->set_drag_forwarding(Callable(), callable_mp(this, &EditorDockManager::_dock_tab_drag_hover).bind(tab_bar), Callable());
```

Comment:
This breaks tab drag and drop because of how tab bar and tab container handle it.
- related https://github.com/godotengine/godot-proposals/issues/8887

Reply:
I can add `tab_hovered` signal to handle that.
But now I wonder if switch on hover should be built-in in the TabBar. That was my original idea, but I went with more local approach.

Reply:
I think its a common enough feature that it can be added to TabBar. We might want it for scene tab bar in the future too.
I can also look into fixing #8887